src/modules/rubberband/filter_rbpitch.cpp

```diff
@@ -66,29 +66,62 @@ static int rbpitch_get_audio( mlt_frame frame, void **buffer, mlt_audio_format *
 	int error = mlt_frame_get_audio( frame, buffer, format, frequency, channels, samples );
 	if ( error ) return error;
 
+	// Make sure the audio is in the correct format
+	// This is useful if the filter is encapsulated in a producer and does not
+	// have a normalizing filter before it.
+	if (*format != mlt_audio_float && frame->convert_audio != NULL)
+	{
+		frame->convert_audio( frame, buffer, format, mlt_audio_float );
+	}
+
+	// Detect the last frame
 	bool last_frame = false;
 	if ( mlt_filter_get_position( filter, frame ) + 1 == mlt_filter_get_length2( filter, frame ) )
 	{
 		// This is the last mlt frame. Drain the rest of the stretcher
 		last_frame = true;
 	}
 
+	// Sanity check parameters
+	// rubberband library crashes have been seen with a very large scale factor
+	// or very small sampling frequency. Very small scale factor and very high
+	// sampling frequency can result in too much audio lag.
+	// Disallow these extreme scenarios for now. Maybe it will be improved in
+	// the future.
+	double pitchscale = mlt_properties_get_double( unique_properties, "pitchscale" );
+	if (pitchscale > 50)
+	{
+		pitchscale = 50;
+	}
+	else if ( pitchscale < 0.05 )
+	{
+		pitchscale = 0.05;
+	}
```

Comment:
Please use the CLAMP macro that we have in mlt_types.h. Also, `50` is an integer literal; please change to `50.0` to make it a double literal.

Reply:
done